crowbar_framework/app/controllers/api/restart_management_controller.rb

```diff
@@ -81,19 +82,28 @@ def restarts_get
 
   def restarts_post
     params.require(:node)
-    params.require(:service)
 
     node_name = params[:node]
+    cookbook = params[:cookbook]
     service = params[:service]
 
     node = get_node_or_raise(node_name)
 
-    managed_cookbooks.each do |cookbook|
-      next unless node.key? :crowbar_wall
-      next unless node[:crowbar_wall].key? :requires_restart
-      next unless node[:crowbar_wall][:requires_restart].key? cookbook
-      next unless node[:crowbar_wall][:requires_restart][cookbook].key? service
-      node[:crowbar_wall][:requires_restart][cookbook].delete(service)
+    if cookbook.nil?
+      node[:crowbar_wall].delete(:requires_restart)
+      Rails.logger.info("restart_management : node #{node_name} cleaned")
+    else
+      raise Crowbar::Error::NotFound unless managed_cookbooks.include? cookbook
+
+      if service.nil?
+        node[:crowbar_wall][:requires_restart].delete(cookbook)
+        Rails.logger.info("restart_management: cookbook #{cookbook} flags cleaned in node "\
+          "#{node_name}")
+      else
+        node[:crowbar_wall][:requires_restart][cookbook].delete(service).nil?
```

Comment:
No need for `nil?`, I think.

Reply:
Also I believe you need to check if `node[:crowbar_wall][:requires_restart][cookbook]` exists, which may be why you had the `nil?` bit (for the test that got dropped)